src/ImageSharp/IO/ChunkedMemoryStream.cs

```diff
@@ -94,247 +57,133 @@ public override long Position
+            bool moveToNextChunk = false;
+            MemoryChunk chunk = this.memoryChunkBuffer[this.currentChunk];
+            int n = bytesToRead;
+            int remainingBytesInCurrentChunk = chunk.Length - this.currentChunkIndex;
+            if (n >= remainingBytesInCurrentChunk)
             {
-                // Exit if no more chunks are currently available
-                if (this.readChunk.Next is null)
-                {
-                    break;
-                }
-
-                this.readChunk = this.readChunk.Next;
-                this.readOffset = 0;
-                chunkBuffer = this.readChunk.Buffer;
-                chunkSize = this.readChunk.Length;
-                if (this.readChunk.Next is null)
-                {
-                    chunkSize = this.writeOffset;
-                }
+                n = remainingBytesInCurrentChunk;
+                moveToNextChunk = true;
             }
 
-            int readCount = Math.Min(count, chunkSize - this.readOffset);
-            chunkBuffer.Slice(this.readOffset, readCount).CopyTo(buffer[offset..]);
-            offset += readCount;
-            count -= readCount;
-            this.readOffset += readCount;
-            bytesRead += readCount;
+            // Read n bytes from the current chunk
+            chunk.Buffer.Memory.Span.Slice(this.currentChunkIndex, n).CopyTo(buffer.Slice(offset, n));
+            bytesToRead -= n;
+            offset += n;
+            bytesRead += n;
+
+            if (moveToNextChunk)
+            {
+                this.currentChunkIndex = 0;
+                this.currentChunk++;
+            }
+            else
+            {
+                this.currentChunkIndex += n;
+            }
         }
 
+        this.position += bytesRead;
         return bytesRead;
     }
 
     /// <inheritdoc/>
-    [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    public override int ReadByte()
+    public override void WriteByte(byte value)
     {
         this.EnsureNotDisposed();
-
-        if (this.readChunk is null)
-        {
-            if (this.memoryChunk is null)
-            {
-                return 0;
-            }
-
-            this.readChunk = this.memoryChunk;
-            this.readOffset = 0;
-        }
-
-        IMemoryOwner<byte> chunkBuffer = this.readChunk.Buffer;
-        int chunkSize = this.readChunk.Length;
-        if (this.readChunk.Next is null)
-        {
-            chunkSize = this.writeOffset;
-        }
-
-        if (this.readOffset == chunkSize)
-        {
-            // Exit if no more chunks are currently available
-            if (this.readChunk.Next is null)
-            {
-                return -1;
-            }
-
-            this.readChunk = this.readChunk.Next;
-            this.readOffset = 0;
-            chunkBuffer = this.readChunk.Buffer;
-        }
-
-        return chunkBuffer.GetSpan()[this.readOffset++];
+        MemoryMarshal.Write(this.singleByteBuffer, ref value);
```

Comment:
ditto `stackalloc`; or possibly even the more exotic:

```
byte b = 0;
var span = MemoryMarshal.CreateSpan(ref b, 1);
```

Reply:
```c#
/// <inheritdoc/>
public override void WriteByte(byte value)
    => this.Write(MemoryMarshal.CreateSpan(ref value, 1));
```